### PR TITLE
refactor(io): format test names

### PR DESCRIPTION
**Pull request by javihernant**
ref #3754

### CODE REVIEW

**Comment by javihernant** [comment]
The approach I am following is, I am trying to get a general idea of what the test is doing and then try to come up with an appropriate name. That is if I feel like the current title is not clear enough. However, I feel like I may have made some mistakes, so maybe it would be good if someone could check. Also, is it neccessary to include the function's name that is being tested. Some of the current title just explain what is going on without using the actual name of the function. And, there are still 19 test files left, each with several tests inside, so maybe this could be done by more people.

File: io/buf_reader_test.ts [hunk 1]
```diff
@@ -138,7 +138,7 @@ Deno.test("bufioBufReader", async function () {
   }
 });
 
-Deno.test("bufioBufferFull", async function () {
+Deno.test("BufReader fails reading with full buffer", async function () {
```
Comment (timreichen) [inline io/buf_reader_test.ts:141]:
```suggestion
Deno.test("BufReader throws if reading with full buffer", async function () {
```

File: io/buf_reader_test.ts [hunk 2]
```diff
@@ -185,7 +185,7 @@ Deno.test("bufioReadString", async function () {
   }
 });
 
-Deno.test("bufReaderReadFull", async function () {
+Deno.test("readFull() fails if array length is too big", async function () {
```
Comment (timreichen) [inline io/buf_reader_test.ts:188]: dito

File: io/buf_reader_test.ts [hunk 3]
```diff
@@ -357,12 +357,12 @@ async function testReadLine(input: Uint8Array) {
...
 }
 
-Deno.test("bufioReadLine", async function () {
+Deno.test("readLine()", async function () {
   await testReadLine(testInput);
   await testReadLine(testInputrn);
 });
 
-Deno.test("bufioReadLineBadResource", async () => {
+Deno.test("readLine() fails with BadResource", async () => {
```
Comment (timreichen) [inline io/buf_reader_test.ts:365]: dito

File: io/buffer_test.ts [hunk 4]
```diff
@@ -86,15 +86,15 @@ function repeat(c: string, bytes: number): Uint8Array {
...
+Deno.test("new Buffer()", () => {
   init();
   assert(testBytes);
   assert(testString);
   const buf = new Buffer(testBytes.buffer);
   check(buf, testString);
 });
 
-Deno.test("bufferBasicOperations", async () => {
+Deno.test("Buffer basic operations", async () => {
```
Comment (timreichen) [inline io/buffer_test.ts:97]:
```suggestion
Deno.test("Buffer handles basic operations", async () => {
```

File: io/buffer_test.ts [hunk 5]
```diff
@@ -133,7 +133,7 @@ Deno.test("bufferBasicOperations", async () => {
   }
 });
 
-Deno.test("bufferReadEmptyAtEOF", async () => {
+Deno.test("read() empty at eof", async () => {
```
Comment (timreichen) [inline io/buffer_test.ts:136]:
```suggestion
Deno.test("read() handles empty at eof", async () => {
```

File: io/buffer_test.ts [hunk 6]
```diff
@@ -142,7 +142,7 @@ Deno.test("bufferReadEmptyAtEOF", async () => {
   assertEquals(result, 0);
 });
 
-Deno.test("bufferLargeByteWrites", async () => {
+Deno.test("write() large bytes", async () => {
```
Comment (timreichen) [inline io/buffer_test.ts:145]:
```suggestion
Deno.test("write() handles large bytes", async () => {
```

File: io/buffer_test.ts [hunk 7]
```diff
@@ -271,7 +271,7 @@ Deno.test({
   },
 });
 
-Deno.test("bufferLargeByteReads", async () => {
+Deno.test("write() large bytes", async () => {
```
Comment (timreichen) [inline io/buffer_test.ts:274]:
```suggestion
Deno.test("write() handles large bytes", async () => {
```

File: io/buffer_test.ts [hunk 8]
```diff
@@ -284,12 +284,12 @@ Deno.test("bufferLargeByteReads", async () => {
   check(buf, "");
 });
 
-Deno.test("bufferCapWithPreallocatedSlice", () => {
+Deno.test("Buffer capacity with preallocated slice", () => {
```
Comment (timreichen) [inline io/buffer_test.ts:287]:
```suggestion
Deno.test("Buffer handles capacity with preallocated slice", () => {
```

**Comment by iuioiua** [comment]
@javihernant, Tim's suggestions are correct. Can you please apply them?

**Comment by javihernant** [comment]
Yea, sure!

File: io/buffer_test.ts [hunk 9]
```diff
@@ -133,7 +133,7 @@ Deno.test("bufferBasicOperations", async () => {
   }
 });
 
-Deno.test("bufferReadEmptyAtEOF", async () => {
+Deno.test("read() handles empty at eof", async () => {
```
Comment (iuioiua) [inline io/buffer_test.ts:136]:
```suggestion
Deno.test("Buffer.read() handles empty at eof", async () => {
```

File: io/buffer_test.ts [hunk 10]
```diff
@@ -142,7 +142,7 @@ Deno.test("bufferReadEmptyAtEOF", async () => {
   assertEquals(result, 0);
 });
 
-Deno.test("bufferLargeByteWrites", async () => {
+Deno.test("write() handles large bytes", async () => {
```
Comment (iuioiua) [inline io/buffer_test.ts:145]:
```suggestion
Deno.test("Buffer.write() handles large bytes", async () => {
```

File: io/buffer_test.ts [hunk 11]
```diff
@@ -153,7 +153,7 @@ Deno.test("bufferLargeByteWrites", async () => {
   check(buf, "");
 });
 
-Deno.test("bufferTooLargeByteWrites", async () => {
+Deno.test("grow() throws if growing beyond maximum size", async () => {
```
Comment (iuioiua) [inline io/buffer_test.ts:156]:
```suggestion
Deno.test("Buffer.grow() throws if growing beyond maximum size", async () => {
```

File: io/buffer_test.ts [hunk 12]
```diff
@@ -171,7 +171,7 @@ Deno.test("bufferTooLargeByteWrites", async () => {
 });
 
 Deno.test({
-  name: "bufferGrowWriteMaxBuffer",
+  name: "writeSync() grows buffer to max capacity",
```
Comment (iuioiua) [inline io/buffer_test.ts:174]:
```suggestion
  name: "Buffer.writeSync() grows buffer to max capacity",
```

File: io/buffer_test.ts [hunk 13]
```diff
@@ -194,7 +194,7 @@ Deno.test({
 });
 
 Deno.test({
-  name: "bufferGrowReadCloseMaxBufferPlus1",
+  name: "readFrom() rejects if read max_size + 1",
```
Comment (iuioiua) [inline io/buffer_test.ts:197]:
```suggestion
  name: "Buffer.readFrom() rejects if read max_size + 1",
```

File: io/buffer_test.ts [hunk 14]
```diff
@@ -211,7 +211,7 @@ Deno.test({
 });
 
 Deno.test({
-  name: "bufferGrowReadSyncCloseMaxBufferPlus1",
+  name: "readFromSync() throws if read max_size + 1",
```
Comment (iuioiua) [inline io/buffer_test.ts:214]:
```suggestion
  name: "Buffer.readFromSync() throws if read max_size + 1",
```

File: io/buffer_test.ts [hunk 15]
```diff
@@ -228,7 +228,7 @@ Deno.test({
 });
 
 Deno.test({
-  name: "bufferGrowReadSyncCloseToMaxBuffer",
+  name: "readFromSync() reads close to max_size",
```
Comment (iuioiua) [inline io/buffer_test.ts:231]:
```suggestion
  name: "Buffer.readFromSync() reads close to max_size",
```

File: io/buffer_test.ts [hunk 16]
```diff
@@ -243,7 +243,7 @@ Deno.test({
 });
 
 Deno.test({
-  name: "bufferGrowReadCloseToMaxBuffer",
+  name: "readFrom() reads close to max_size",
```
Comment (iuioiua) [inline io/buffer_test.ts:246]:
```suggestion
  name: "Buffer.readFrom() reads close to max_size",
```

File: io/buffer_test.ts [hunk 17]
```diff
@@ -257,7 +257,7 @@ Deno.test({
 });
 
 Deno.test({
-  name: "bufferReadCloseToMaxBufferWithInitialGrow",
+  name: "readFrom() reads close to max_size with initial grow",
```
Comment (iuioiua) [inline io/buffer_test.ts:260]:
```suggestion
  name: "Buffer.readFrom() reads close to max_size with initial grow",
```

File: io/buffer_test.ts [hunk 18]
```diff
@@ -271,7 +271,7 @@ Deno.test({
   },
 });
 
-Deno.test("bufferLargeByteReads", async () => {
+Deno.test("write() handles large bytes", async () => {
```
Comment (iuioiua) [inline io/buffer_test.ts:274]:
```suggestion
Deno.test("Buffer.write() handles large bytes", async () => {
```

File: io/buffer_test.ts [hunk 19]
```diff
@@ -284,12 +284,12 @@ Deno.test("bufferLargeByteReads", async () => {
...
 });
 
-Deno.test("bufferCapWithPreallocatedSlice", () => {
+Deno.test("Buffer handles capacity with preallocated slice", () => {
   const buf = new Buffer(new ArrayBuffer(10));
   assertEquals(buf.capacity, 10);
 });
 
-Deno.test("bufferReadFrom", async () => {
+Deno.test("readFrom()", async () => {
```
Comment (iuioiua) [inline io/buffer_test.ts:292]:
```suggestion
Deno.test("Buffer.readFrom()", async () => {
```

File: io/buffer_test.ts [hunk 20]
```diff
@@ -311,7 +311,7 @@ Deno.test("bufferReadFrom", async () => {
   });
 });
 
-Deno.test("bufferReadFromSync", async () => {
+Deno.test("readFromSync()", async () => {
```
Comment (iuioiua) [inline io/buffer_test.ts:314]:
```suggestion
Deno.test("Buffer.readFromSync()", async () => {
```

File: io/buffer_test.ts [hunk 21]
```diff
@@ -333,7 +333,7 @@ Deno.test("bufferReadFromSync", async () => {
   });
 });
 
-Deno.test("bufferTestGrow", async () => {
+Deno.test("grow()", async () => {
```
Comment (iuioiua) [inline io/buffer_test.ts:336]:
```suggestion
Deno.test("Buffer.grow()", async () => {
```

File: io/buffer_test.ts [hunk 22]
```diff
@@ -357,7 +357,7 @@ Deno.test("bufferTestGrow", async () => {
   }
 });
 
-Deno.test("testBufferBytesArrayBufferLength", () => {
+Deno.test("writeAllSync() and readFromSync()", () => {
```
Comment (iuioiua) [inline io/buffer_test.ts:360]:
```suggestion
Deno.test("Buffer.write() and Buffer.readFromSync()", () => {
```
Note: `writeAll()` is not a `Buffer` method.

File: io/buffer_test.ts [hunk 23]
```diff
@@ -376,7 +376,7 @@ Deno.test("testBufferBytesArrayBufferLength", () => {
   }
 });
 
-Deno.test("testBufferBytesCopyFalse", () => {
+Deno.test("bytes() with copy set to false", () => {
```
Comment (iuioiua) [inline io/buffer_test.ts:379]:
```suggestion
Deno.test("Buffer.bytes() with copy set to false", () => {
```

File: io/buffer_test.ts [hunk 24]
```diff
@@ -391,7 +391,7 @@ Deno.test("testBufferBytesCopyFalse", () => {
   assert(actualBytes.buffer.byteLength > actualBytes.byteLength);
 });
 
-Deno.test("testBufferBytesCopyFalseGrowExactBytes", () => {
+Deno.test("bytes() with copy set to false after grow()", () => {
```
Comment (iuioiua) [inline io/buffer_test.ts:394]:
```suggestion
Deno.test("Buffer.bytes() with copy set to false after Buffer.grow()", () => {
```

File: io/buf_writer_test.ts [hunk 25]
```diff
@@ -9,7 +9,7 @@ import { StringWriter } from "./string_writer.ts";
 import { bufsizes } from "./_test_common.ts";
 import type { Writer, WriterSync } from "./types.ts";
 
-Deno.test("bufioWriter", async function () {
+Deno.test("write() writes different number of bytes", async function () {
```
Comment (iuioiua) [inline io/buf_writer_test.ts:12]:
```suggestion
Deno.test("BufWriter.write()", async function () {
```

File: io/buf_writer_test.ts [hunk 26]
```diff
@@ -42,7 +42,7 @@ Deno.test("bufioWriter", async function () {
   }
 });
 
-Deno.test("bufioWriterSync", function () {
+Deno.test("writeSync()", function () {
```
Comment (iuioiua) [inline io/buf_writer_test.ts:45]:
```suggestion
Deno.test("BufWriter.writeSync()", function () {
```

File: io/buf_reader_test.ts [hunk 27]
```diff
@@ -104,14 +104,14 @@ const bufreaders: NamedBufReader[] = [
   // { name: "lines", fn: readLines },
 ];
 
-Deno.test("bufioReaderSimple", async function () {
+Deno.test("readBytes() reads string from buffer", async function () {
```
Comment (iuioiua) [inline io/buf_reader_test.ts:107]:
```suggestion
Deno.test("BufReader.readBytes() reads string from buffer", async function () {
```

File: io/buf_reader_test.ts [hunk 28]
```diff
@@ -138,7 +138,7 @@ Deno.test("bufioBufReader", async function () {
   }
 });
 
-Deno.test("bufioBufferFull", async function () {
+Deno.test("BufReader throws when reading with full buffer", async function () {
```
Comment (iuioiua) [inline io/buf_reader_test.ts:141]:
```suggestion
Deno.test("BufReader.readSlice() throws when reading with full buffer", async function () {
```

File: io/buf_reader_test.ts [hunk 29]
```diff
@@ -160,7 +160,7 @@ Deno.test("bufioBufferFull", async function () {
   assertEquals(actual, "world!");
 });
 
-Deno.test("bufioReadString", async function () {
+Deno.test("readString()", async function () {
```
Comment (iuioiua) [inline io/buf_reader_test.ts:163]:
```suggestion
Deno.test("BufReader.readString()", async function () {
```

File: io/buf_reader_test.ts [hunk 30]
```diff
@@ -185,7 +185,7 @@ Deno.test("bufioReadString", async function () {
   }
 });
 
-Deno.test("bufReaderReadFull", async function () {
+Deno.test("readFull() throws if array length is too big", async function () {
```
Comment (iuioiua) [inline io/buf_reader_test.ts:188]:
```suggestion
Deno.test("BufReader.readFull() throws if array length is too big", async function () {
```

File: io/buf_reader_test.ts [hunk 31]
```diff
@@ -212,7 +212,7 @@ Deno.test("bufReaderReadFull", async function () {
   }
 });
 
-Deno.test("bufioPeek", async function () {
+Deno.test("peek()", async function () {
```
Comment (iuioiua) [inline io/buf_reader_test.ts:215]:
```suggestion
Deno.test("BufReader.peek()", async function () {
```

File: io/buf_reader_test.ts [hunk 32]
```diff
@@ -357,12 +357,12 @@ async function testReadLine(input: Uint8Array) {
   }
 }
 
-Deno.test("bufioReadLine", async function () {
+Deno.test("readLine()", async function () {
```
Comment (iuioiua) [inline io/buf_reader_test.ts:360]:
```suggestion
Deno.test("BufReader.readLine()", async function () {
```

File: io/buf_reader_test.ts [hunk 33]
```diff
@@ -357,12 +357,12 @@ async function testReadLine(input: Uint8Array) {
...
   await testReadLine(testInput);
   await testReadLine(testInputrn);
 });
 
-Deno.test("bufioReadLineBadResource", async () => {
+Deno.test("readLine() throws with BadResource", async () => {
```
Comment (iuioiua) [inline io/buf_reader_test.ts:365]:
```suggestion
Deno.test("BufReader.readLine() throws with BadResource", async () => {
```

File: io/buf_reader_test.ts [hunk 34]
```diff
@@ -371,7 +371,7 @@ Deno.test("bufioReadLineBadResource", async () => {
   }, Deno.errors.BadResource);
 });
 
-Deno.test("bufioReadLineBufferFullError", async () => {
+Deno.test("readLine() fills a full buffer", async () => {
```
Comment (iuioiua) [inline io/buf_reader_test.ts:374]:
```suggestion
Deno.test("BufReader.readLine() fills a full buffer", async () => {
```

**Review by iuioiua — CHANGES_REQUESTED** [review]
Sorry, there's actually a number of nits I missed in my previous comment.

**Comment by javihernant** [comment]
Okay, Iḿ changing them now!

**Comment by iuioiua** [comment]
@javihernant, please resolve any conversations in the PR and mark as ready for review once ready.

File: io/buffer_test.ts [hunk 35]
```diff
@@ -86,15 +86,15 @@ function repeat(c: string, bytes: number): Uint8Array {
   return ui8;
 }
 
-Deno.test("bufferNewBuffer", () => {
+Deno.test("new Buffer()", () => {
```
Comment (iuioiua) [inline io/buffer_test.ts:89]:
```suggestion
Deno.test("Buffer", () => {
```

**Review by iuioiua — APPROVED** [review]
LGTM. Thank you.

**Pull request merged**